src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -290,31 +299,25 @@
        (IllegalArgumentException.
          "'distinct_resources' query parameter requires accompanying parameters 'distinct_start_time' and 'distinct_end_time'")))))
 
-(defn warn-experimental
-  [entity warn-experimental]
-  (when (and warn-experimental
-             (contains? experimental-entities entity))
-    (log/warn (format
-                "The %s endpoint is experimental and may be altered or removed in the future."
-                (name entity)))))
-
 (defn query-route
-  "Returns a route for querying PuppetDB that supports the standard
-   paging parameters. Also accepts GETs and POSTs. Composes
-   `optional-handlers` with the middleware function that executes the
-   query."
-  [entity version param-spec & optional-handlers]
-  (let [handlers (or optional-handlers [identity])]
-    (app
-      (extract-query param-spec)
-      (apply comp
-             (fn [{:keys [params globals puppetdb-query]}]
-               (warn-experimental entity (:warn-experimental globals))
-               (produce-streaming-body
-                 entity
-                 version
-                 (validate-distinct-options! (merge (keywordize-keys params) puppetdb-query))
-                 (:scf-read-db globals)
-                 (:url-prefix globals)
-                 (:pretty-print globals)))
-             optional-handlers))))
+  ([version param-spec]
+   (query-route version param-spec [identity]))
+  ([version param-spec optional-handlers]
+   (app
+    (extract-query param-spec)
+    (apply comp
+           (fn [{:keys [params globals puppetdb-query]}]
+             (produce-streaming-body version
+                                     (validate-distinct-options! (merge (keywordize-keys params) puppetdb-query))
+                                     (select-keys globals [:scf-read-db :url-prefix :pretty-print :warn-experimental])))
+           optional-handlers))))
+
+(defn query-route-from
+  "Convenience wrapper for query-route, which automatically wraps the query in a
+  `from` to set context."
+  ([entity version param-spec]
+   (query-route-from entity version param-spec [identity]))
+  ([entity version param-spec optional-handlers]
+   (let [handlers (cons (partial restrict-query-to-entity entity)
+                        optional-handlers)]
+     (query-route version param-spec handlers))))
```

Comment:
Any way that query-route and query-route-from could be merged, or could the `([version param-spec] (query-route version param-spec [identity]))` case of query-route be removed, just a little bothered by the duplication of the optional-handlers defaulting interface?